src/dolphin/_log.py

```diff
@@ -0,0 +1,106 @@
+"""Exports a get_log function which sets up easy logging.
+
+Uses the standard python logging utilities, just provides
+nice formatting out of the box across multiple files.
+
+Usage:
+
+    from ._log import get_log
+    logger = get_log()
+
+    logger.info("Something happened")
+    logger.warning("Something concerning happened")
+    logger.error("Something bad happened")
+    logger.critical("Something just awful happened")
+    logger.debug("Extra printing we often don't need to see.")
+    # Custom output for this module:
+    logger.success("Something great happened: highlight this success")
+"""
+import logging
+import time
+from logging import Formatter
+
+COLORS = False
+
+__all__ = ["get_log", "log_runtime"]
+
+
+def get_log(debug=False, name="dolphin.log"):
+    """Create a nice log format for use across multiple files.
+
+    Default logging level is INFO
+
+    Parameters
+    ----------
+    name : Optional[str]
```

Comment:
This syntax for describing the type of `name` might be a little confusing.

`Optional[str]` and `str, optional` mean two different things to me when they appear in the docstring like this.

* `Optional[str]` makes me think that you mean "optional" in the sense of [`typing.Optional`](https://docs.python.org/3/library/typing.html#typing.Optional), i.e. this parameter can be either a `str` or `None`.

   If I was trying to describe this in a docstring, I would probably just use

   ```
   name : str or None
       ...
   ```

* `str, optional`, on the other hand, says to me that this is an optional argument of type `str` (i.e. a `str` argument with a default value). This is the syntax that the [numpy docstring guidelines](https://numpydoc.readthedocs.io/en/latest/format.html#parameters) suggest for describing optional arguments.

I think the second meaning is what you were going for.

Not sure if you agree with my interpretation of the syntax, but it seems like there does need to be some way to disambiguate these two common uses of "optional".